src/rlang/vec-lgl.c

```diff
@@ -1,24 +1,25 @@
 #include "rlang.h"
+#include <stdlib.h>
 
 r_ssize r_lgl_sum(r_obj* x, bool na_true) {
   if (r_typeof(x) != R_TYPE_logical) {
     r_abort("Internal error: Excepted logical vector in `r_lgl_sum()`");
   }
 
-  r_ssize n = r_length(x);
+  const r_ssize n = r_length(x);
+  const int* v_x = r_lgl_cbegin(x);
 
+  // This can't overflow since `sum` is necessarily smaller or equal
+  // to the vector length expressed in `r_ssize`
   r_ssize sum = 0;
-  const int* p_x = r_lgl_cbegin(x);
 
-  for (r_ssize i = 0; i < n; ++i) {
-    // This can't overflow since `sum` is necessarily smaller or equal
-    // to the vector length expressed in `r_ssize`.
-    int x_i = p_x[i];
-
-    if (na_true && x_i) {
-      sum += 1;
-    } else if (x_i == 1) {
-      sum += 1;
+  if (na_true) {
+    for (r_ssize i = 0; i < n; ++i) {
+      sum += (bool) v_x[i];
+    }
+  } else {
+    for (r_ssize i = 0; i < n; ++i) {
+      sum += (v_x[i] == 1);
     }
   }
```

Comment:
Branching off `na_true` early, and avoiding branching in the loop, instead letting `+= 0` be a fast no-op when we don't hit a value that increments the counter